qrenderdoc/Code/qrenderdoc.cpp

```diff
@@ -291,6 +295,27 @@ int main(int argc, char *argv[])
               .arg(configFilename));
     }
 
+    int replayHostIndex = -1;
+    if(parser.isSet(replayhost))
+    {
+      QString replayHost = parser.value(replayhost);
+      for(int i = config.RemoteHosts.count() - 1; i >= 0; i--)
+      {
+        if(QString(config.RemoteHosts[i]->hostname) == replayHost)
+        {
+          replayHostIndex = i;
+          break;
+        }
+      }
+      if(replayHostIndex < 0)
```

Comment:
Can you explain the reasoning behind adding the remote host here? Normally this kind of change happens in the UI itself explicitly so the user is in full control.

I'm not opposed to the idea necessarily but I'd like to hear the use-case this is intended to solve.

Reply:
Our use-case is that we launch qrenderdoc through another application/script. That application figures out the ip address of a remote server we want to connect to, and it wants some way to pass that info to qrenderdoc and presets the replay context when launching qrenderdoc. We don't want our users to go through the manually adding a new remote host via remote manager dialogue window every time there is a new remote server, so I choose to add a new entry in RemoteHost list if the hostname is never seen before.